src/config.rs

```diff
@@ -354,6 +365,16 @@ fn parse_color_string(color_string: &str) -> Option<ansi_term::Color> {
         return Some(Color::Fixed(ansi_color_num));
     }
 
+    // Check palette for a matching user-defined color
+    if let Some(palette_color) = palette.as_ref().and_then(|x| x.get(color_string)) {
+        log::trace!(
+            "Read user-defined color string: {} defined as {}",
+            color_string,
+            palette_color
+        );
+        return parse_color_string(palette_color, palette);
```

Comment:
To avoid potential stack overflows (this behavior should be documented):
```suggestion
        return parse_color_string(palette_color, None);
```